src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -0,0 +1,1414 @@
+        }
+
+        private class Class_With_NullableUInt64_And_Float
+        {
+            public ulong? NullableUInt64Number { get; set; }
+            [JsonInclude]
+            public List<float> FloatNumbers;
+        }
+
+        [Fact]
+        public static void Number_AsKeyValuePairValue_RoundTrip()
+        {
+            var obj = new KeyValuePair<ulong?, List<float>>(JsonNumberTestData.NullableULongs.LastOrDefault(), JsonNumberTestData.Floats);
+
+            // Serialize
+            string serialized = JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr);
+
+            // Deserialize
+            obj = JsonSerializer.Deserialize<KeyValuePair<ulong?, List<float>>>(serialized, s_optionReadAndWriteFromStr);
+
+            // Test roundtrip
+            JsonTestHelper.AssertJsonEqual(serialized, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        [Fact]
+        public static void Number_AsObjectWithParameterizedCtor_RoundTrip()
+        {
+            var obj = new MyClassWithNumbers(JsonNumberTestData.NullableULongs.LastOrDefault(), JsonNumberTestData.Floats);
+
+            // Serialize
+            string serialized = JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr);
+
+            // Deserialize
+            obj = JsonSerializer.Deserialize<MyClassWithNumbers>(serialized, s_optionReadAndWriteFromStr);
+
+            // Test roundtrip
+            JsonTestHelper.AssertJsonEqual(serialized, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        private class MyClassWithNumbers
+        {
+            public ulong? Ulong { get; }
+            public List<float> ListOfFloats { get; }
+
+            public MyClassWithNumbers(ulong? @ulong, List<float> listOfFloats)
+            {
+                Ulong = @ulong;
+                ListOfFloats = listOfFloats;
+            }
+        }
+
+        [Fact]
+        public static void Number_AsObjectWithParameterizedCtor_PropHasAttribute()
+        {
+            string json = @"{""ListOfFloats"":[""1""]}";
+            // Strict handling on property overrides loose global policy.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json, s_optionReadFromStr));
+
+            // Serialize
+            json = @"{""ListOfFloats"":[1]}";
+            MyClassWithNumbers_PropsHasAttribute obj = JsonSerializer.Deserialize<MyClassWithNumbers_PropsHasAttribute>(json);
+
+            // Number serialized as JSON number due to strict handling on property which overrides loose global policy.
+            Assert.Equal(json, JsonSerializer.Serialize(obj, s_optionReadAndWriteFromStr));
+        }
+
+        private class MyClassWithNumbers_PropsHasAttribute
+        {
+            [JsonNumberHandling(JsonNumberHandling.Strict)]
+            public List<float> ListOfFloats { get; }
```

Comment:
Question: Presumably strict is identical in behavior to not putting any annotation at all, correct?

Is the only reason to put this attribute value is to override the global options settings?

Reply:
Yes.